app/src/main/java/com/github/wnder/HistoryActivity.java

```diff
@@ -24,13 +48,73 @@ protected void onCreate(Bundle savedInstanceState) {
         //Set layout
         setContentView(R.layout.activity_history);
 
+        //Buttons to cycle between the images
+        left_button = findViewById(R.id.left_history);
+        right_button = findViewById(R.id.right_history);
+
         //set placeholder
         image = findViewById(R.id.historyImage);
-        image.setImageURI(Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag));
+        //image.setImageURI(Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag));
         //When clicked open activity for specific picture history
         image.setOnClickListener((view) -> {
             Intent intent = new Intent(this, PictureHistoryActivity.class);
             startActivity(intent);
         });
+
+        getUserPictures();
+    }
+
+    private void setupButtons() {
+        if(pictureList == null){
+            throw new IllegalArgumentException();
+        }
+        if(pictureList.isEmpty()){
+            //TODO: add default image or just nothing
+        }
+        else{
+            pictures_max_number = pictureList.size() - 1;
+            picture_index = 0;
+
+            //Set the first image to be displayed
+            setImage(0);
+
+            left_button.setOnClickListener(view ->{
+                picture_index = picture_index == 0 ? pictures_max_number : picture_index - 1;
+                setImage(picture_index);
+            });
+
+            right_button.setOnClickListener(view ->{
+                picture_index = picture_index == pictures_max_number ? 0 : picture_index + 1;
+                setImage(picture_index);
+            });
+        }
+    }
+
+    //Set the image, both for imageView and for the existingPicture
+    private void setImage(int index){
+        if(index < 0 || index > pictures_max_number){
+            throw new IllegalArgumentException();
+        }
+
+        pictureList.get(index).onBitmapAvailable(bmp ->image.setImageBitmap(bmp));
+        pictureDisplayed = pictureList.get(index);
+    }
+
+    private void getUserPictures(){
+        List<ExistingPicture> picsList = new ArrayList<>();
+        User user = GlobalUser.getUser();
+
+        if(user instanceof SignedInUser) { //We can search for the pictures of the signed in user on the online database
```

Comment:
Perhaps move `onGuessedPicturesAvailable` to the `User` class with a blanked implementation for now (will be added with the local database). Then in `SignedInUser` we can use the current code with a fallback to the inherited function in case there is no internet connection.

Reply:
I have to see with the others if this is consistent with what we already have. But you're right, the "instanceof" seems a bit off here.